contracts/BoostedMasterChefJoe.sol

```diff
@@ -0,0 +1,621 @@
+            .div(1000)
+            .div(ACC_TOKEN_PRECISION)
+            .sub(user.rewardDebt)
+            .add(claimableJoe[_pid][_user]);
+
+        // If it's a double reward farm, we return info about the bonus token
+        if (address(pool.rewarder) != address(0)) {
+            bonusTokenAddress = address(pool.rewarder.rewardToken());
+            bonusTokenSymbol = IERC20(pool.rewarder.rewardToken()).safeSymbol();
+            pendingBonusToken = pool.rewarder.pendingTokens(_user);
+        }
+    }
+
+    /// @notice Update reward variables for all pools. Be careful of gas spending!
+    /// @param pids Pool IDs of all to be updated. Make sure to update all active pools.
+    function massUpdatePools(uint256[] calldata pids) external {
+        uint256 len = pids.length;
+        for (uint256 i = 0; i < len; ++i) {
+            updatePool(pids[i]);
+        }
+    }
+
+    /// @notice Calculates and returns the `amount` of JOE per block.
+    function joePerSec() public view returns (uint256 amount) {
+        uint256 total = 1000;
+        uint256 lpPercent = total.sub(MASTER_CHEF_V2.devPercent()).sub(MASTER_CHEF_V2.treasuryPercent()).sub(
+            MASTER_CHEF_V2.investorPercent()
+        );
+        uint256 lpShare = MASTER_CHEF_V2.joePerSec().mul(lpPercent).div(total);
+        amount = lpShare.mul(MASTER_CHEF_V2.poolInfo(MASTER_PID).allocPoint).div(MASTER_CHEF_V2.totalAllocPoint());
+    }
+
+    /// @notice Update reward variables of the given pool.
+    /// @param pid The index of the pool. See `poolInfo`.
+    function updatePool(uint256 pid) public {
+        PoolInfo memory pool = poolInfo[pid];
+        if (block.timestamp > pool.lastRewardTimestamp) {
+            uint256 lpSupply = pool.lpToken.balanceOf(address(this));
+            if (lpSupply > 0) {
+                uint256 secondsElapsed = block.timestamp.sub(pool.lastRewardTimestamp);
+                uint256 joeReward = secondsElapsed.mul(joePerSec()).mul(pool.allocPoint).div(totalAllocPoint);
+                pool.accJoePerShare = pool.accJoePerShare.add(
+                    (joeReward.mul(ACC_TOKEN_PRECISION).div(pool.totalBoostedAmount))
+                );
+            }
+            pool.lastRewardTimestamp = block.timestamp;
+            poolInfo[pid] = pool;
+            emit UpdatePool(pid, pool.lastRewardTimestamp, lpSupply, pool.accJoePerShare);
+        }
+    }
+
+    /// @notice Deposit LP tokens to BMCJ for JOE allocation.
+    /// @param pid The index of the pool. See `poolInfo`.
+    /// @param amount LP token amount to deposit.
+    function deposit(uint256 pid, uint256 amount) external nonReentrant {
+        harvestFromMasterChef();
+        updatePool(pid);
+        PoolInfo storage pool = poolInfo[pid];
+        UserInfo storage user = userInfo[pid][msg.sender];
+
+        if (user.amount > 0) {
+            // Harvest JOE
+            uint256 boostedAmount = user.amount.mul(getUserBoost(pid, msg.sender)).div(1000);
```

Comment:
same question regarding where the `1000` is coming from, should we store it as a variable so we can re-use it across the contract?

Reply:
see comment above